packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -90,16 +94,33 @@ export const GridColumnHeaderItem = ({
     [apiRef, column.field],
   );
 
+  const publishClick = React.useCallback(
+    (eventName: string) => (event: React.MouseEvent) => {
+      apiRef!.current.setCellFocus({
```

Comment:
you should treat focus separately
Header and cell are separate as well

Reply:
Need to rename the method `setFocus`

Reply:
I can rename it to `setFocus` but I think it makes more sense to treat column headers and cells the same, after all they are all cells but with just different purposes.

Reply:
I see your point but I would prefer to treat them separately. 
I have thought about that in the past. But I came to the result that it won't scale and the code will be more hacky. As rowIndex===null or some kind of if... It also won't have the same parameters...
So I would like to keep the same approach, but for `ColumnHeader` and then call setFocus

Reply:
Alright, I'll look at it. I'm more concerned about the way I currently used to handle the `rowIndex`. I'm using `null` for the column header cells. It is working and it looks clean but I would like another opinion. The issue is that I still need it and the actual grid rows start from `0` which is ok. I also don't think that the column headers should have `rowIndex` of `0`.

Reply:
Updated